src/main/java/org/gridsuite/study/server/service/StudyService.java

```diff
@@ -1447,42 +1450,38 @@ private void updateStatuses(UUID studyUuid, UUID nodeUuid, List<ModificationInfo
             case EQUIPMENT_ATTRIBUTE_MODIFICATION: {
                 // for now only one sub type (switch), but after need to extract the equipment type to select the right update type
                 Set<String> substationIds = getSubstationIds(modifications);
-                notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY, substationIds);
+                notificationService.emitNetworkImpacts(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY, NetworkImpcatsNotificationPayload.builder().impactedSubstationsIds(substationIds).build());
                 updateStatuses(studyUuid, nodeUuid);
-                notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_SWITCH);
+                notificationService.emitNetworkImpacts(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_SWITCH, NetworkImpcatsNotificationPayload.builder().build());
                 break;
             }
             case BRANCH_STATUS_MODIFICATION: {
                 Set<String> substationIds = getSubstationIds(modifications);
-                notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY, substationIds);
+                notificationService.emitNetworkImpacts(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY, NetworkImpcatsNotificationPayload.builder().impactedSubstationsIds(substationIds).build());
                 updateStatuses(studyUuid, nodeUuid);
-                notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_LINE);
+                notificationService.emitNetworkImpacts(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_LINE, NetworkImpcatsNotificationPayload.builder().build());
                 break;
             }
-            case EQUIPMENT_DELETION: {
-                modifications.stream()
-                    .map(EquipmentDeletionInfos.class::cast)
-                    .forEach(deletionInfo ->
-                        notificationService.emitStudyEquipmentDeleted(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY, deletionInfo.getSubstationIds(),
-                            deletionInfo.getEquipmentType(), deletionInfo.getEquipmentId()));
-                updateStatuses(studyUuid, nodeUuid);
-                break;
-            }
-            case LINE_SPLIT_WITH_VOLTAGE_LEVEL: {
-                Set<String> allImpactedSubstationIds = modifications.stream()
-                        .map(ModificationInfos::getSubstationIds).flatMap(Set::stream).collect(Collectors.toSet());
-                List<EquipmentModificationInfos> deletions = modifications.stream()
+            default: {
+                Set<EquipmentModificationInfos> deletions = modifications.stream()
```

Comment:
Why `Set` ?
If you want to remove duplications add `@EqualsAndHashCode` to the `EquipmentDeletionNotification` class

                Set<EquipmentDeletionNotification> deletionsInfos = modifications.stream()
                    .filter(modif -> modif.getType() == ModificationType.EQUIPMENT_DELETION)
                    .map(EquipmentDeletionInfos.class::cast)
                    .map(d -> new EquipmentDeletionNotification(d.getEquipmentId(), d.getEquipmentType()))
                    .collect(Collectors.toSet());

                // we need to filter out the substation ids that are deleted
                Set<String> deletionsIds = deletionsInfos.stream().map(EquipmentDeletionNotification::getId).collect(Collectors.toSet());
                Set<String> impactedSubstationIds = getSubstationIds(modifications).stream()
                    .filter(id -> !deletionsIds.contains(id))
                    .collect(Collectors.toSet());

                notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY,
                    NetworkImpcatsNotificationPayload
                        .builder()
                        .impactedSubstationsIds(impactedSubstationIds)
                        .deletedEquipments(deletionsInfos)
                        .build());
                updateStatuses(studyUuid, nodeUuid);